### PR TITLE
Use `emojilib` to enhance results

### LINKED ISSUE #30: Enhance 3rd party result with local information
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/emoj/issues/30)

Some emoji's just aren't findable with the 3rd party service. For instance https://github.com/sindresorhus/alfred-emoj/issues/14. What would it take to enhance the results?

Could we bundle a local db to add emoji that are hard to find or just aren't in the 3rd party service?

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $80.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($80.00)
### Submitted pull Requests
- [#39 solves #30](https://issuehunt.io/r/sindresorhus/emoj/pull/39)
- [#40 Add `emojilib` to enhance results](https://issuehunt.io/r/sindresorhus/emoj/pull/40)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/emoj/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by stroncium**
Adds `emojilib` results to what we get from getdango.

Filtering of `emojilib` is done with simple heuristic(ignore some characters, do exact matches on long searches, substring matches on short searches).

Also, allowed using only `emojilib` if getdango API is not available for some reason.

Fixes #30 
Closes #40

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,3 +12,49 @@ module.exports = async input => {
 
 	return response.body.results.map(x => x.text);
 };
+
+const getEmojilibEmojis = input => {
+	const reSource = input.toLowerCase().split(/\s/g)
+		.map(v => v.replace(/\W/g, ''))
+		.filter(v => v.length > 0)
+		.map(v => (v.length >= 4) ? v : '^' + v + '$')
```
Comment (sindresorhus) [inline index.js]: Can you add a code comment about why 4 was chosen and why the check is there?

**Comment by sindresorhus** [comment]
This generally looks good :)

**Comment by sindresorhus** [comment]
There's a check in `ui.js` that checks if dango is offline. I don't think we need that anymore since we have this.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by stroncium** [comment]
@sindresorhus I think I cleaned up `ui.js` though I have no idea what it is used for/from so didn't test.

File: index.js [hunk 2]
```diff
@@ -12,11 +12,14 @@ const getGetdangoEmojis = async input => {
 	return results.map(result => result.text);
 };
 
+// This value was picked experimentally
```
Comment (sindresorhus) [inline index.js]: Based on what? Would be nice with a more elaborate reasoning.

**Comment by sindresorhus** [comment]
I tried turning off my wifi, and it doesn't work:

```
› horse



GotError: getaddrinfo ENOTFOUND emoji.getdango.com emoji.getdango.com:80
    at onError (/Users/sindresorhus/dev/oss/emoj/node_modules/got/dist/source/request-as-event-emitter.js:140:29)
    at handleRequest (/Users/sindresorhus/dev/oss/emoj/node_modules/got/dist/source/request-as-event-emitter.js:173:17)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:56:26)
```

**Comment by stroncium** [comment]
@sindresorhus Ok, figured it out now, didn't notice there was interactive mode. Getdango errors were thrown up in case local db got no results, ignoring them completely now.

**Pull request merged**